testsuites/CBLTester/CBL_Functional_tests/TestSetup_FunctionalTests/test_custom_conflicts.py

```diff
@@ -0,0 +1,1013 @@
+            db.updateDocument(cbl_db, doc_id=doc_id, data=data)
+
+    repl_config = replicator.configure(cbl_db, sg_blip_url, continuous=True, channels=channels,
+                                       replicator_authenticator=replicator_authenticator,
+                                       replication_type=replicator_type, conflict_resolver="delete_not_win")
+    repl = replicator.create(repl_config)
+    replicator.start(repl)
+    replicator.wait_until_replicator_idle(repl)
+    total = replicator.getTotal(repl)
+    completed = replicator.getCompleted(repl)
+    replicator.stop(repl)
+    assert total == completed, "total is not equal to completed"
+
+    # printing doc content before replication conflicted docs
+    try:
+        sg_docs_content = sg_client.get_bulk_docs(sg_url, sg_db, doc_ids, session)[0]
+    except Exception as err:
+        log_info("Requesting for deleted docs from SG: {}".format(err))
+        sg_docs_content = []
+    cbl_docs = db.getDocuments(cbl_db, doc_ids)
+    if replicator_type == "pull":
+        assert sg_docs_content == [], "CCR failed to resolved conflict with delete_not_win and replicated content to SG"
+        assert cbl_docs, "CCR failed to resolved conflict with delete_not_win"
+    elif replicator_type == "push_pull":
+        for sg_doc in sg_docs_content:
+            doc_id = sg_doc["_id"]
+            cbl_doc = cbl_docs[doc_id]
+            assert sg_doc["sg_new_update1"] == cbl_doc["sg_new_update1"], "CCR failed to resolve conflict " \
+                                                                          "with delete_not_win"
+            assert sg_doc["sg_new_update2"] == cbl_doc["sg_new_update2"], "CCR failed to resolve conflict " \
+                                                                          "with delete_not_win"
+            assert sg_doc["sg_new_update3"] == cbl_doc["sg_new_update3"], "CCR failed to resolve conflict " \
+                                                                          "with delete_not_win"
+            assert "random" not in sg_doc, "CCR failed to resolve conflict with delete_not_win"
+            assert "cbl_random" in sg_doc, "CCR failed to resolve conflict with delete_not_win"
+
+
+@pytest.mark.listener
+@pytest.mark.custom_conflict
+@pytest.mark.replication
+@pytest.mark.parametrize("replicator_type", [
+    "pull",
+    "push_pull"
+])
+def test_exception_thrown_custom_conflicts(params_from_base_test_setup, replicator_type):
+    """
+    @summary: resolve conflicts as per local doc
+    1. Create few docs in app and get them replicated to SG. Stop the replication once docs are replicated.
+    2. Update docs couple of times with different updates on one side and delete on other side.
+    This will create conflict.
+    3. Start the replication with delete_not_wins CCR algorithm
+    4. Verifies that CBL has retains its changes. For push and pull replication SG changes should be override with
```

Comment:
this is not the right verification. It is different from spreadsheet.

Reply:
Actually the verification steps are not the only way to verify that scenario. The exception is a user thrown exception, which will lead to no additional information or measure. The idea is to ensure that app is not crashed. I did check with Jim about it, he told the scenario is not needed and doesn't even add any additional value. Please discuss with other devs and let me know your thought

Reply:
done